src/powersystem.jl

```diff
@@ -60,15 +62,194 @@ struct PowerNetwork
         Ybus, Yf_br, Yt_br, Yf_tr, Yt_tr = makeYbus(data)
 
         # bus type indexing
-        ref, pv, pq = bustypeindex(bus, gen)
-
+        ref, pv, pq, bustype = bustypeindex(bus, gen)
+    
         Sbus = assembleSbus(gen, load, SBASE, nbus)
+        
+        new(V, Ybus, data, nbus, ngen, nload, bustype, ref, pv, pq,
+            Sbus)
+    end
+
+end
+
+"""
+    view(PowerNetwork)
+
+Prints power network characteristics.
 
-        new(V, Ybus, data, nbus, ngen, nload, ref, pv, pq, Sbus)
+"""
+
+function view(pf::PowerNetwork)
+    println("Power Network characteristics:")
+    @printf("\tBuses: %d. Slack: %d. PV: %d. PQ: %d\n", pf.nbus, length(pf.ref),
+            length(pf.pv), length(pf.pq))
+    println("\tGenerators: ", pf.ngen, ".")
+    println("\tLoads: ", pf.nload, ".")
+
+    # Print system status
+    @printf("\t==============================================\n")
+    @printf("\tBUS \t TYPE \t VMAG \t VANG \t P \t Q\n")
+    @printf("\t==============================================\n")
+    
+    for i=1:pf.nbus
+        type = pf.bustype[i]
+        vmag = abs(pf.V[i])
+        vang = angle(pf.V[i])*(360.0/pi)
+        pinj = real(pf.Sbus[i])
+        qinj = imag(pf.Sbus[i])
+        @printf("\t%i \t  %d \t %1.3f\t%3.2f\t%3.3f\t%3.3f\n", i,
+                type, vmag, vang, pinj, pinj)
     end
+
 end
 
 
+
+
+"""
+    assemble_vecs(PowerNetwork)
```

Comment:
I think we should think about a more explicit naming. How about `partition`? Or split in three functions `state`, `control`, `variable`? Or just overloading a function `get` so that we just have to type `get(pf::PowerNetwork, State())`?

Reply:
@frapac Ill do the get overloading as you proposed!